google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java

```diff
@@ -237,16 +248,26 @@ private void showWarnings() {
     if (!isValidConfigurationFile(appYaml.getText())) {
       errorMessage.setText(GctBundle.getString("appengine.deployment.error.staging.yaml"));
       showError = true;
-    }
-
-    try {
-      if (isRuntimeCustom() && !isValidConfigurationFile(dockerfile.getText())) {
-        errorMessage.setText(GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
+    } else {
+      try {
+        if (isRuntimeCustom()) {
```

Comment:
may change depending on if/how you handle @patflynn's comment here, but as is its a little hard to follow this code. Could it be refactored for clarity (extracting named methods where appropriate etc.)?